praw/models/reddit/subreddit.py

```diff
@@ -7,12 +7,23 @@
 from io import StringIO
 from json import dumps, loads
 from os.path import basename, dirname, isfile, join
-from typing import List
+from typing import (
+    List,
+    TYPE_CHECKING,
+    Optional,
+    Dict,
+    Any,
+    Union,
+    Iterator,
+    Generator,
+    Set,
+)
 from urllib.parse import urljoin
 from xml.etree.ElementTree import XML
 
 import websocket
 from prawcore import Redirect
+from requests import Response
```

Comment:
It looks like `Response` is only used in a type signature. Should this be under `if TYPE_CHECKING:`?

I'm not very familiar with the intricacies of Python's `typing`, so I'm just curious.

Reply:
I imported Response because it doesn't interfere with the import chain. I put under `if TYPE_CHECKING` anything that could cause a circular import loop, or anything else that I imported from PRAW. However, I highly doubt a third-party package would form such a loop.